ldraw_to_scad.py

```diff
@@ -1,266 +1,303 @@
+                result.append(f"function {intfile_name} = [")
+        if len(params) >= 2 and params[1] == 'NOFILE':
+            intfile = self.getdummy()
+            result.append("];")
+            intfile_name = LDrawConverter.make_function_name(intfile)
+            result.append(f"function {intfile_name} = [")
+
+    def convert_line(self, part_line):
+        """ Translate a single line. """
+        stripped = part_line.rstrip()
+        result = [f"// {stripped}"] if self.settings['commented'] else []
+        params = stripped.split(maxsplit=14)
+        if not params:
+            return result
+        if params[0] == "0":
+            self.convert_line_0(result, params, stripped)
+        elif params[0] == "1":
+            self.adddep(params[14])
+            result.append(
+                f"  [{','.join(params[:14])}, "
+                f"{LDrawConverter.make_function_name(params[14])}],")
+        elif params[0] in ["2", "3", "4", "5"]:
+            outparams = params[:{'2': 8, '3': 11, '4': 14, '5': 14}[params[0]]]
+            result.append(f"  [{','.join(outparams)}],")
+        return result
+
+    @staticmethod
+    def include(comp, path):
+        """ Generate use statement. """
+        relpath = (
+            os.path.join('openscad', *comp) if path == '/' else
+            os.path.relpath(os.path.join(*comp), path))
+        return f'use <{relpath}.scad>'
+
+    def process_lines(self, name, path, lines):
+        """ Translate all lines of a file. """
+        self.filedep = [set(), set()]
+        self.mpd_main = None
+        result = []
+        for line in lines:
+            result.extend(self.convert_line(line))
+        function_name = LDrawConverter.make_function_name(name)
+        if self.settings['selfcontained']:
+            for file in self.getdeps():
+                self.enqueue(file, path)
+            result = [f"function {function_name} = ["] + result + ["];"]
+        else:
+            result = [LDrawConverter.include(['lib'], path)] + \
+                     [LDrawConverter.include(self.find_part(f), path)
+                      for f in sorted(self.getdeps())] + \
+                     [f"function {function_name} = ["] + result + ["];"] + \
+                     [f"makepoly({function_name}, "
+                      f"line={self.settings['line']});"]
+        if self.mpd_main and self.mpd_main != name:
+            main_function = LDrawConverter.make_function_name(self.mpd_main)
+            result.append(f"function {main_function} = {function_name};\n")
+        return result
+
+    def enqueue(self, name, path=None, ldrfile=None, scadfile=None):
+        """ enqueue a file to be processed """
+        if name not in self.queue[1]:
+            if not ldrfile:
+                lpath, base = self.find_part(name)
+            self.queue[0][name] = [
+                path if path else lpath,
+                ldrfile if ldrfile else
+                os.path.join(self.library_root, lpath, base) +
```

Comment:
Can you use indentation to show where these statements continue?

Reply:
I think I made this a bit nicer now. BTW: I have also a patch available to run pylint and pycodestyle along with pytest. Do you want me to add this one as well to avoid degradation in those aspects in the future?

Reply:
That would be great - I was considering putting pylint in.